src/test/vat.t.sol

```diff
@@ -645,35 +644,101 @@ contract BiteTest is DSTest {
         assertEq(cat.litter(), 0);
         uint auction = cat.bite("gold", address(this));
-        // (,,,,,,,uint tab) = flip.bids(auction);
-        // assertTrue(rad(75 ether) - cat.litter() < ray(3 ether));
-        // assertEq(cat.litter(), tab);
+
         assertEq(ink("gold", address(this)), 50 ether);
         assertEq(art("gold", address(this)), 75 ether);
         assertEq(vow.sin(now), rad(75 ether));
         assertEq(gem("gold", address(this)), 900 ether);
 
-        assertEq(vat.balanceOf(address(this)), 150 ether);
-        assertEq(vat.balanceOf(address(vow)),    0 ether);
-        flip.tend(auction, 50 ether, rad( 1 ether));
+        assertEq(vat.dai(address(this)),  rad(150 ether));
+        assertEq(vat.dai(address(vow)),     rad(0 ether));
+        flip.tend(auction, 50 ether, rad(1 ether));
         assertEq(cat.litter(), rad(75 ether));
-        assertEq(vat.balanceOf(address(this)), 149 ether);
+        assertEq(vat.dai(address(this)), rad(149 ether));
         flip.tend(auction, 50 ether, rad(75 ether));
-        assertEq(vat.balanceOf(address(this)), 75 ether);
+        assertEq(vat.dai(address(this)), rad(75 ether));
 
         assertEq(gem("gold", address(this)),  900 ether);
         flip.dent(auction, 25 ether, rad(75 ether));
         assertEq(cat.litter(), rad(75 ether));
-        assertEq(vat.balanceOf(address(this)), 75 ether);
+        assertEq(vat.dai(address(this)), rad(75 ether));
         assertEq(gem("gold", address(this)), 925 ether);
         assertEq(vow.sin(now), rad(75 ether));
 
         hevm.warp(now + 4 hours);
         flip.deal(auction);
         assertEq(cat.litter(), 0);
         assertEq(gem("gold", address(this)),  950 ether);
-        assertEq(vat.balanceOf(address(this)), 75 ether);
-        assertEq(vat.balanceOf(address(vow)),  75 ether);
+        assertEq(vat.dai(address(this)),   rad(75 ether));
+        assertEq(vat.dai(address(vow)),    rad(75 ether));
+    }
+
+    // tests a partial lot liquidation because it would fill the literbox
+    function test_partial_litterbox_realistic_values() public {
+        // spot = tag / (par . mat)
+        // tag=5, mat=2
+        vat.file("gold", 'spot', ray(2.5 ether));
+        vat.frob("gold", me, me, me, 100 ether, 150 ether);
+
+        // tag=4, mat=2
+        vat.file("gold", 'spot', ray(1 ether));  // now unsafe
+        cat.file("gold", "chop", 1.13 ether);
+
+        assertEq(ink("gold", address(this)), 100 ether);
+        assertEq(art("gold", address(this)), 150 ether);
+        assertEq(vow.Woe(), 0 ether);
+        assertEq(gem("gold", address(this)), 900 ether);
+
+        // To check this yourself, use the following rate calculation (example 8%):
+        //
+        // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
+        uint256 EIGHT_PCT = 1000000002440418608258400030;
+        jug.file("gold", "duty", EIGHT_PCT);
+        hevm.warp(now + 10 days);
+        jug.drip("gold");
```

Comment:
Just FYI, the easy way to get a non-zero `rate` in a test is to call `Vat.fold` directly since you have `auth` superpowers as the test contract, but it's fine to do it this way.